lib/azure/armrest/configuration.rb

```diff
@@ -202,7 +202,9 @@ def token_expiration
       # argument to 'USGov'. Otherwise, set it to nil.
       #
       def environment=(env)
-        fetch_token if env != environment
+        return if env == environment
+        set_auth_and_resource_urls(env)
+        fetch_token
```

Comment:
We should make `environment=` a private method. Then there is no need to `fetch_token` here since `environment=` method is  is called only once. We can also merge `set_auto_and_resource_urls` back to this method since there is no other logic